### PR TITLE
Add escape for strings in filterwarnings (#2303)

### PR DESCRIPTION
**Pull request by amarjandu**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `DCP/1` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Ran `make requirements_update` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or added `no sandbox` label</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit
- [x] Moved linked issue to Merged
- [ ] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Unassign reviewer from PR

### CODE REVIEW

File: test/azul_test_case.py [hunk 1]
```diff
@@ -29,36 +32,42 @@ class AzulTestCase(TestCase):
...
+                '"@coroutine" decorator is deprecated since Python 3.8, use "async def" instead'
             },
             UserWarning: {
-                r'https://github\.com/DataBiosphere/azul/issues/2114',
+                'https://github.com/DataBiosphere/azul/issues/2114',
             }
         }
         for warning_class, message_patterns in permitted_warnings_.items():
             for message_pattern in message_patterns:
+                if type(message_pattern) is not Pat:
```
Comment (jessebrennan) [inline test/azul_test_case.py:69]: I'm not sure if there's anything wrong with checking a type this way, but the common pattern---both in our code and in my experiences with Python---is to use `isinstance()` instead. 
```suggestion
                if not isinstance(message_pattern, Pat):
```

File: test/azul_test_case.py [hunk 2]
```diff
@@ -29,36 +32,42 @@ class AzulTestCase(TestCase):
     @classmethod
     def setUpClass(cls) -> None:
         super().setUpClass()
+
+        class Pat(str):
```
Comment (jessebrennan) [inline test/azul_test_case.py:36]: Consider the name `RE` instead of `Pat`. I think it's more familiar.

**Review by jessebrennan — CHANGES_REQUESTED** [review]
Looks like unit tests are failing. Glancing over the changes, the code looks good.

Just a couple suggestions.

File: test/azul_test_case.py (hunk 1 above)
Comment (amarjandu) [reply test/azul_test_case.py:69]: good point, ill change this, also https://stackoverflow.com/a/1549854/2205587

**Comment by amarjandu** [comment]
@jessebrennan i think the failure in the drop! commit is preventing the pipeline for the coveralls to run, ill talk to hannes after standup as he requested the drop! commit to showcase how the warnings.filterwarnings is working. 

The commit that carries your requested changes passes :)

**Review by jessebrennan — APPROVED** [review]
> he requested the drop! commit to showcase how the warnings.filterwarnings is working

Ah, I should have looked more carefully. Looks good then!

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
Will remove drop commit now.

**Comment by hannes-ucsc** [comment]
For demo showcase changes and repeat the experiment from the drop! commit.

**Pull request merged**